### PR TITLE
drivers: modem: wnc-m14a2a: make sure interface is up

### PR DESCRIPTION
**Pull request by mike-scott**
The modem initialization can take quite a while and there are times
when the offload_get() function can be called while it's still in
progress.  Let's add a wait there to make sure the interface is up.

Signed-off-by: Michael Scott <mike@foundries.io>

### CODE REVIEW

**Comment by mike-scott** [comment]
@jukkar @nashif @pfalcon sorry for the late patch.

We were unable to come up with a better solution (for now) to make sure interfaces are "ready", so my stop-gap is to insert a wait here prior to getting handing back the completed net_context.

Let me know if this is ok for 1.13.

**Comment by pfalcon** [comment]
So, this hits again on the same matter as https://github.com/zephyrproject-rtos/zephyr/pull/9314, https://github.com/zephyrproject-rtos/zephyr/issues/9330 . Please remind again why you can't rely on net_config_init()'s timeout (explicitly or thru CONFIG_NET_CONFIG_INIT_TIMEOUT).

All in all, I'm happy to approve, but I'm not a decision power at all, and to me this looks like a "feature", not a "critical bugfix/trivial non-code fix", which should be kind of stuff merged after RC2. So perhaps, release managers would like to know more about motivation to get this into 1.13.

**Comment by mike-scott** [comment]
@pfalcon Not a feature at all.   The modem specifically disables the wait in net-app layer so that the kernel can progress and user sees output in the terminal (init can take from 30 seconds to 1 minute).

Without this change, apps will start and attempt to send data before interface is ready.

**Comment by pfalcon** [comment]
> The modem specifically disables the wait in net-app layer so that the kernel can progress and user sees output in the terminal (init can take from 30 seconds to 1 minute).

So, I guess you should decide what's more important so far: user seeing output, or having reliable app startup *without adding hacky-hacky code*.

> Without this change, apps will start and attempt to send data before interface is ready.

Ok, with this change, app will start up, send data, then modem will go down randomly (out of cell coverage, whatever), but app will still attempt to send data. So, this is a crude hack. At least, not on the level of generic network service ;-), so let me approve it. I hope to add my comments to #9330 soon.

**Review by pfalcon — APPROVED** [review]

**Review by jukkar — APPROVED** [review]
This is quite hackish. Preferably we should let the modem to indicate when it is ready, and after that bring the interface up. Could this be done like that?

I am not sure if this could go into 1.13, the release is suppose to happen on Monday and only bug fixes with high priority are allowed.

**Comment by mike-scott** [comment]
@jukkar Yes, this is a hack until a better method is available.  The modem actually does bring the interface up when it's ready (as well as bring it down during reset):
https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/modem/wncm14a2a.c#L1424

**Comment by jukkar** [comment]
> The modem actually does bring the interface up when it's ready

As the driver knows when it is ready so serve things, why just return -EAGAIN or similar and let the application to decide what to do next. Waiting the interface UP event in this case is a bit problematic as we might loose the event and perhaps the application would like to do something else while the modem is initializing.

**Pull request closed without merging**
